gobblin-service/src/main/java/gobblin/service/modules/core/GobblinServiceManager.java

```diff
@@ -319,9 +343,13 @@ public void stop() throws ApplicationException {
   @VisibleForTesting
   void connectHelixManager() {
     try {
-      this.helixManager.connect();
-      this.helixManager.getMessagingService().registerMessageHandlerFactory(
-          Message.MessageType.USER_DEFINE_MSG.toString(), getUserDefinedMessageHandlerFactory());
+      if (this.helixManager.isPresent()) {
```

Comment:
Check is already done before calling this method, so it can be removed from one of the places.